substratevm/src/com.oracle.svm.core/src/com/oracle/svm/core/jfr/JfrManager.java

```diff
@@ -81,9 +82,18 @@ public static JfrManager get() {
         return ImageSingletons.lookup(JfrManager.class);
     }
 
-    public RuntimeSupport.Hook startupHook() {
+    public static RuntimeSupport.Hook initializationHook() {
+        /* Parse arguments early on so that we can tear down the isolate more easily if it fails. */
         return isFirstIsolate -> {
-            parseFlightRecorderLogging(SubstrateOptions.FlightRecorderLogging.getValue());
+            parseFlightRecorderLogging();
+            parseFlightRecorderOptions();
+        };
+    }
+
+    public static RuntimeSupport.Hook startupHook() {
+        return isFirstIsolate -> {
+            parseFlightRecorderLogging();
+            parseFlightRecorderOptions();
```

Comment:
Why do have to do parse again here if we already did it in the previous hook?

Reply:
Thanks, I removed it.